aikit/automl/controller.py

```diff
@@ -0,0 +1,287 @@
+            try:
+                next_job = next(self._block_search_iterator)
+                return self.get_job('block-search', *next_job)
+            except StopIteration:
+                logger.info("Block-search iterator cleared out")
+                self._block_search_iterator_empty = True
+
+        p_exploration = self.find_exploration_proba()
+        logger.debug('Exploration probability: {:.2%}'.format(p_exploration))
+
+        if self.random_state.rand(1)[0] <= p_exploration:
+            return self.get_job('exploration', *self.generator.draw_random_graph())
+        else:
+            return self.get_guided_job()
+
+    def get_job(self, job_type, graph, all_models_params, blocks_to_use):
+        code = convert_graph_to_code(graph, all_models_params, also_returns_mapping=True)
+        job_id = md5_hash(code['json_code'])
+        return {
+            'job_id': job_id,
+            'job_type': job_type,
+            'job_creation_time': datetime.datetime.strftime(datetime.datetime.now(), "%Y-%m-%d %H:%M:%S"),
+            'graph': {'nodes': list(graph.nodes), 'edges': list(graph.edges)},
+            'all_models_params': all_models_params,
+            'blocks_to_use': blocks_to_use,
+            'model_code': code['json_code'],
+            'name_mapping': code['name_mapping']
+        }
+
+    def find_exploration_proba(self):
+        """ function to retrieve the probability to run a random model, proba decrease with number of model tested """
+        n_jobs = self.reader.get_number_of_successful_jobs()
+        if n_jobs <= self.automl_guider.min_nb_of_models:
+            return 1
+        elif n_jobs <= 100:
+            return 0.5
+        elif n_jobs <= 500:
+            return 0.25
+        else:
+            return 0.1
+
+    def get_guided_job(self):
+        """ Guided job : draw several models and guess which is best
+
+        1) Mean + 2 * Std
+        benchmark = metric_prediction + 2 * np.sqrt(metric_variance_prediction)
+        2) Proba new >= best =>
+        (Mean - Best) / Std
+        3) E(New * (1[new >= best])
+        Rmk : si on utilise des rang, best presque 1
+        ii = np.argmax(benchmark)
+
+        NOTE: on va prendre a peu pres le plus best avec une proba 1/4 (suivant les tests)
+        # On peut aussi faire une heuristic en suppossant benchmark uniform (pas tres loin de la verite vu qu'on a fitter un rank...)
+
+        # Comme ca je prend pas l'argmax, mais quelque chose d'un peu plus exploratoir
+        # ... peut etre que argmax ca marcherait mieux (surement plus petite variation autour du meilleurs model)
+
+        # On peut aussi virer les trucs vraiment pas bon...
+        # Ou tirer au hasard parmis les meilleurs ..
+
+        TODO : on peut faire descendre la temperature en court de route.... a peu pres equivalent à gérer l'explortion...
+        """
+        jobs = [self.get_job('guided', *self.generator.draw_random_graph()) for _ in range(100)]
+
```

Comment:
while we are refactoring, maybe put all that logic into the guider  ... makes more sens to me now. So maybe keep : 

`
def get_guided_jobs(self):
    jobs = [self.get_job('guided', *self.generator.draw_random_graph()) for _ in range(100)]
    return self.automl_guider.choise_best_job(jobs)
`

Reply:
Makes lots of sense!